### PR TITLE
Extract analytics to helper module, fix lost events on page redirects, and add a few new events.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This PR extracts our little Google Analytics helper code into a [standalone npm package](https://github.com/DoSomething/analytics).

It also updates `dosomething_helpers_add_analytics_event` to flash analytics events to the session, and inject the analytics script itself when rendering a page, which fixes an issue where any events that were followed by `drupal_goto`s would be lost. (This will fix the `Authentication:Login` and `Authentication:Register` events that weren't firing.)

Finally, it removes some confusing events and adds a couple of new ones for tracking events that happen in the authentication flow: `Link:Click:Login`, `Link:Click:Register`, `Link:Click:Signup (Authenticated)`, and `Link:Click:Signup (Unauthenticated)`.

#### How should this be reviewed?
The interesting stuff happens in `dosomething_helpers_add_analytics_event` and `preprocess_html.inc`. Most of the rest is just plopping attributes into arrays.

Definitely check out [the source](https://github.com/DoSomething/analytics/blob/56a0e52a32cec86a001ac172f2f6d721000ad13c/analytics.js) for the extracted Analytics package and let me know how it looks!

#### Any background context you want to provide?
💃

#### Relevant tickets
🍃

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love
- [ ] Post a message in #phoenix if this includes something that causes a rebuild!

### CODE REVIEW

**Comment by DFurnes** [comment]
[As an admin, I want to add GA events to Phoenix so that I can understand how users interact with our login modal.](https://trello.com/c/qLTqk5XP/163-as-an-admin-i-want-to-add-ga-events-to-phoenix-so-that-i-can-understand-how-users-interact-with-our-login-modal)

**Comment by weerd** [comment]
> This PR extracts our little Google Analytics helper code into a standalone npm package.

Huzzah! Glad this finally happened 💃

File: lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module [hunk 1]
```diff
@@ -561,30 +561,23 @@ function dosomething_helpers_floor_decimal($value, $decimals = 1) {
...
+function dosomething_helpers_add_analytics_event($category, $action, $label = NULL) {
+  $_SESSION['dosomething_analytics_events'][] = [$category, $action, $label];
 }
 
 /**
  * Creates a simple query encoded string with basic or customized UTM params
  * that can be added to any url.
  * @param  $nid
  * @param  array $options - custom utm values to use.
+ * @return string
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module:580]: 😼

File: lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module [hunk 2]
```diff
@@ -561,30 +561,23 @@ function dosomething_helpers_floor_decimal($value, $decimals = 1) {
...
-  if ($label) {
-    $event_string = $event_string . ', ' . $label;
-  }
-  if ($value) {
-    $event_string = $event_string  . ',' . $value;
-  }
-  $google_analytics_js = 'if(typeof(ga) !== "undefined" && ga !== null) { ga("send", "event", ' . $event_string . '); }';
-  drupal_add_js($google_analytics_js, 'inline');
+function dosomething_helpers_add_analytics_event($category, $action, $label = NULL) {
+  $_SESSION['dosomething_analytics_events'][] = [$category, $action, $label];
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module:572]: Nice! Good idea 💡

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc [hunk 3]
```diff
@@ -35,28 +35,29 @@ function dosomething_signup_forms($form_id, $args) {
...
-      'class' => array(
-        'button',
-      ),
-    ),
-  );
+    '#attributes' => [
+      'class' => ['button'],
+      'data-track-category' => 'Link',
+      'data-track-action' => 'Click',
+      'data-track-label' => 'Signup (Authenticated)',
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.forms.inc:58]: Much nice! Job great! 🐕

File: lib/themes/dosomething/paraneue_dosomething/includes/preprocess/preprocess_html.inc [hunk 4]
```diff
@@ -6,6 +6,14 @@
 function paraneue_dosomething_preprocess_html(&$vars) {
   global $user;
 
+  // Output any queued analytics events from `dosomething_helpers_add_analytics_event`:
+  foreach ($_SESSION['dosomething_analytics_events'] as $index => $event) {
+    $analytics_js = 'analyze("' . implode('", "', array_filter($event)) . '");';
+    drupal_add_js($analytics_js, 'inline');
+
+    unset($_SESSION['dosomething_analytics_events'][$index]);
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/includes/preprocess/preprocess_html.inc:14]: 👍

**Review by weerd — APPROVED** [review]
Looks awesome! 👍

File: lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js [hunk 5]
```diff
@@ -4,97 +4,52 @@
...
-import Modal from 'dosomething-modal';
+const $ = require('jquery');
+const Analytics = require('@dosomething/analytics');
+const Validation = require('dosomething-validation');
+const Modal = require('dosomething-modal');
+
+function init() {
+  // Configure `data-ga-*` and `data-track-*` links.
+  Analytics.init('ga');
+  Analytics.init('track');
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Doesn't it default to `track`? Or do you _need_ to specify it here?

Comment (DFurnes) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Yeah, it does. I just thought it made it a bit more clear why the method was being called twice to explicitly write each namespace. Can change (or better yet, can search and remove old `-ga-` attributes).

Comment (weerd) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Makes sense :) Can leave for now, and maybe do the latter (remove old `-ga-` attributes) in a future cleanup PR? Up to you, but not vital for merge 😄

Comment (DFurnes) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: Just checked and there were only two! 💥

Comment (weerd) [reply lib/themes/dosomething/paraneue_dosomething/js/utilities/Analytics.js]: 👍

**Pull request merged**